fix(autoware_tensorrt_yolox): missing cstdint include and ament_index_cpp test dependency

## Description

`autoware_tensorrt_yolox` fails to compile on Ubuntu 24.04 / ROS 2 Jazzy with two errors:

### 1. Missing `#include <cstdint>` in `label.hpp`

[`label.hpp`](https://github.com/autowarefoundation/autoware_universe/blob/fe12efd70793bd1d5b4e06a20016dd2c9e1803d2/perception/autoware_tensorrt_yolox/include/autoware/tensorrt_yolox/label.hpp) uses `uint32_t` but does not include `<cstdint>`. This causes compilation to fail:

```
include/autoware/tensorrt_yolox/label.hpp:38:33: error: 'uint32_t' has not been declared
```

### 2. Missing `ament_index_cpp` test dependency

[`test/test_label.cpp`](https://github.com/autowarefoundation/autoware_universe/blob/fe12efd70793bd1d5b4e06a20016dd2c9e1803d2/perception/autoware_tensorrt_yolox/test/test_label.cpp) includes `<ament_index_cpp/get_package_share_directory.hpp>` but `ament_index_cpp` is not listed as a `<test_depend>` in [`package.xml`](https://github.com/autowarefoundation/autoware_universe/blob/fe12efd70793bd1d5b4e06a20016dd2c9e1803d2/perception/autoware_tensorrt_yolox/package.xml). This causes the test to fail:

```
test/test_label.cpp:15:10: fatal error: ament_index_cpp/get_package_share_directory.hpp: No such file or directory
```

## Fix

1. Add `#include <cstdint>` to `label.hpp`
2. Add `<test_depend>ament_index_cpp</test_depend>` to `package.xml`

--- a/perception/autoware_tensorrt_yolox/include/autoware/tensorrt_yolox/label.hpp
+++ b/perception/autoware_tensorrt_yolox/include/autoware/tensorrt_yolox/label.hpp
@@ -15,6 +15,7 @@
 #ifndef AUTOWARE__TENSORRT_YOLOX__LABEL_HPP_
 #define AUTOWARE__TENSORRT_YOLOX__LABEL_HPP_
 
+#include <cstdint>
 #include <optional>
 #include <string>
 #include <unordered_map>

--- a/perception/autoware_tensorrt_yolox/package.xml
+++ b/perception/autoware_tensorrt_yolox/package.xml
@@ -36,6 +36,7 @@
 
   <exec_depend>autoware_image_transport_decompressor</exec_depend>
 
+  <test_depend>ament_index_cpp</test_depend>
   <test_depend>ament_lint_auto</test_depend>
   <test_depend>autoware_lint_common</test_depend>
 

